pkg/plugins/resources/k8s/mapper.go

```diff
@@ -0,0 +1,60 @@
+package k8s
+
+import (
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kumahq/kuma/pkg/config/core/resources/store"
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+	k8s_model "github.com/kumahq/kuma/pkg/plugins/resources/k8s/native/pkg/model"
+	util_k8s "github.com/kumahq/kuma/pkg/util/k8s"
+)
+
+type ResourceMapper interface {
+	Map(resource model.Resource, namespace string) (k8s_model.KubernetesObject, error)
+}
+type ResourceMapperFunc func(resource model.Resource, namespace string) (k8s_model.KubernetesObject, error)
+
+func (f ResourceMapperFunc) Map(resource model.Resource, namespace string) (k8s_model.KubernetesObject, error) {
+	return f(resource, namespace)
+}
+
+func NewMapper(systemNamespace string, storeType store.StoreType, kubeFactory KubeFactory) ResourceMapper {
+	return ResourceMapperFunc(func(resource model.Resource, namespace string) (k8s_model.KubernetesObject, error) {
```

Comment:
q: why this pattern?

Why not simple
```
type resourceMapper struct{...deps}

func (r *resourceMapper) Map(resource model.Resource, namespace string) (k8s_model.KubernetesObject, error)

func NewMapper(...) { return &resourceMapper{} }
```

I'd make more sense if you only had func as a type that you use around the code and remove interface
```
type ResourceMapper func(resource model.Resource, namespace string) (k8s_model.KubernetesObject, error)
```

Reply:
It felt to me like a struct with state was superfluous here and only makes things harder to read. `http.StripPrefix` uses this pattern for example.

Reply:
Actually let me rewrite this a bit which might make it more readable

Reply:
@jakubdyszkiewicz PTAL now

Reply:
Hmm... I still don't understand what is the benefit both of having interface and func. If you want to have func, can't we just have
```
type ResourceMapper func(resource model.Resource, namespace string) (k8s_model.KubernetesObject, error)
```
?

For me having this
```
type ResourceMapperFunc func(resource model.Resource, namespace string) (k8s_model.KubernetesObject, error)

func (f ResourceMapperFunc) Map(resource model.Resource, namespace string) (k8s_model.KubernetesObject, error) {
	return f(resource, namespace)
}
```
makes things harder to read.

I looked at `http.StripPrefix` and I actually don't understand why there is both `Handler` interface and `HandlerFunc`.

Reply:
Because some handlers are stateful (needs a struct to put the state) some others are stateless (just make it a function). It's like creating a functional interface in java.

Reply:
In that specific case it's probably unecessary to have an interface as the only 2 implems I can think of end being stateless

Reply:
Removed the interface

Reply:
> Because some handlers are stateful (needs a struct to put the state) some others are stateless (just make it a function). It's like creating a functional interface in java.

Ha, ok. I guess this makes sense. Also it's an interface that is implemented A LOT, so it's worth to have both.